// survey-angular-ui/esm2015/components/popup/popup.component.js
import { Component, Input } from "@angular/core";
import { BaseAngular } from "../../base-angular";
import { createPopupViewModel } from "survey-core";
import * as i0 from "@angular/core";
import * as i1 from "./popup.service";
export class PopupComponent extends BaseAngular {
    constructor(viewContainerRef, changeDetectorRef, popupService) {
        super(changeDetectorRef, viewContainerRef);
        this.popupService = popupService;
    }
    getModel() {
        return this.popupModel;
    }
    onModelChanged() {
        var _a;
        this.destroyModel();
        this.model = createPopupViewModel(this.popupModel, (_a = this.viewContainerRef) === null || _a === void 0 ? void 0 : _a.element.nativeElement.parentElement);
        this.model.initializePopupContainer();
        this.portalHost = this.popupService.createComponent(this.model);
    }
    ngOnInit() {
        this.onModelChanged();
    }
    destroyModel() {
        var _a, _b;
        (_a = this.portalHost) === null || _a === void 0 ? void 0 : _a.detach();
        (_b = this.model) === null || _b === void 0 ? void 0 : _b.unmountPopupContainer();
    }
    ngOnDestroy() {
        super.ngOnDestroy();
        this.destroyModel();
    }
}
PopupComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: PopupComponent, deps: [{ token: i0.ViewContainerRef }, { token: i0.ChangeDetectorRef }, { token: i1.PopupService }], target: i0.ɵɵFactoryTarget.Component });
PopupComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: PopupComponent, selector: "sv-ng-popup, '[sv-ng-popup]'", inputs: { popupModel: "popupModel" }, usesInheritance: true, ngImport: i0, template: "<div></div>", isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: PopupComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "sv-ng-popup, '[sv-ng-popup]'",
                    template: "<div></div>"
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }, { type: i0.ChangeDetectorRef }, { type: i1.PopupService }]; }, propDecorators: { popupModel: [{
                type: Input
            }] } });
